hikvision_tftpd.py

```diff
@@ -102,7 +113,14 @@ def _handshake_read(self):
     def _tftp_read(self):
         pkt, addr = self._tftp_sock.recvfrom(65536)
         now = time.strftime(_TIME_FMT)
-        if pkt.startswith(self._tftp_rrq_prefix):
+        if pkt.startswith(self._tftp_rrq_prefix) and pkt.find(self._tftp_blksize_option) != -1:
```

Comment:
Could we split the options into a dict here first? I think it'd be clearer when reading the code, and we could have better debugging output as well. And it just bugs me a little that it's first doing a purely string find first and then assuming blksize_index is defined and such, even though I know we're unlikely to encounter malformed packets that will cause problems.

Reply:
Yeah I'm less than proud of this part of the code. Python is not regular language so knowing exactly what tools are best for a job allude me and this was the first thing I thought of that worked and again, for the purposes I needed the server for was enough.

Reply:
Fair enough. Realistically speaking, this works fine, and we can always clean it up later.